server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -449,7 +449,7 @@ public boolean canApproximate(SearchContext context) {
         if (context.from() + context.size() == 0) {
             this.setSize(SearchContext.DEFAULT_TRACK_TOTAL_HITS_UP_TO);
         } else {
-            this.setSize(Math.max(context.from() + context.size(), context.trackTotalHitsUpTo() + 1));
+            this.setSize(Math.max(context.from() + context.size(), context.trackTotalHitsUpTo()));
```

Comment:
I think it also worth to change the logic here, we don't need to consider trackTotalHits anymore if we will always rely on shortcut to give that. The size here can just be `context.from() + context.size()`. 

This requires some context on the "approximate" framework and make the decision.

@harshavamsi

Reply:
We have to battle test that. The `shortcutTotalHitCount` today only returns the `numDocs` if its a `MatchAllDocsQuery` query and having a threshold of 1. This way all the collected docs from approximation need not go through the collect method part of TopFieldCollector from Lucene and the competitive iterator can early kick in.

But the decision of just using `context.from() + context.size()` or  `context.trackTotalHitsUpTo()` should be added after doing some tests.

Reply:
But I dont deny just having `context.from() + context.size()` should improve the performance even better.